src/groupByKey.ts

```diff
@@ -1,7 +1,23 @@
 type GroupsMap<T> = {
-  [key: string]: T[];
+  [key: string]: Array<T>;
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: Array<T>,
+  key: string): GroupsMap<T> {
+  const outObj: GroupsMap<T> = {};
+
+  let tempKey: string | number;
+
+  for (let i: number = 0; i < items.length;) {
+    tempKey = items[i][key];
```

Comment:
```suggestion
  for (let i: number = 0; i < items.length;) {
    const tempKey = items[i][key];
```

There is also such an option

Reply:
To save file lines